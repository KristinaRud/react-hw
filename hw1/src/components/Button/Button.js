import { Component } from "react";
import './Button.scss';

class Button extends Component{
    render(){
        const {backgroundColor, text, onClick}=this.props;
        return (
            <>
            <button className="btn" style={{backgroundColor}} onClick={onClick}>{text}</button>
            </>
        )
    }
}

export default Button;