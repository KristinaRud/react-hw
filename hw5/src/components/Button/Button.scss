.btn-buy {
    background-color: rgba(0, 0, 0, 0);
    border-radius: 0;
    border: none;
    box-sizing: border-box;
    cursor: pointer;
    display: inline-block;
    font-weight: bold;
    margin: 15px auto;
    overflow: hidden;
    padding: 0;
    position: relative;
    letter-spacing: 0px;
    width: auto;
    z-index: 10;
}

.btn-buy::before {
    border-width: 1px 1px 0 0;
    right: 0;
    margin-left: 16px;
}

.btn-buy::after {
    border-width: 0 0 1px 1px;
    left: 0;
    margin-right: 16px;
}

.btn-buy::before,
.btn-buy::after {

    background-color: #e62429;
    border-color: #e62429;
    transition: none;

    border-style: solid;
    box-sizing: border-box;
    content: "";
    display: block;
    height: 15px;

}