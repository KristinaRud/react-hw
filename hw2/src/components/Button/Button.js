import { Component } from "react";
import "./Button.scss";
import PropTypes from "prop-types";

class Button extends Component {
  render() {
    const { text, onClick} = this.props;
    return (
      <>
        <button type={type}
          className={className}
          onClick={onClick}
        >
          <span className="innerFill">{text}</span>
        </button>
      </>
    );
  }
}

export default Button;

Button.propTypes = {
  isModal: PropTypes.func,
  currentComics: PropTypes.func,
  text:PropTypes.string,
};

Button.defaultProps={
  type:"button",
  className:"btn-buy"
}