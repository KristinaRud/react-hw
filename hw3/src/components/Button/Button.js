import "./Button.scss";
import PropTypes from "prop-types";

const Button =({ content, onClick, className, type})=> {
  return (
    <>
      <button type={type}
        className={className}
        onClick={onClick}
      >
       {content}
      </button>
    </>
  );    
}

export default Button;

Button.propTypes = {
  isModal: PropTypes.func,
  currentComics: PropTypes.func,
  content:PropTypes.element,
};

Button.defaultProps={
  type:"button",
}