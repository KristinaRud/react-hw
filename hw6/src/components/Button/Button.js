import "./Button.scss";
import PropTypes from "prop-types";

const Button =({ content, onClick, className, type})=> {
  return (
    <>
      <button type={type}
        className={className}
        onClick={onClick}
      >
       {content}
      </button>
    </>
  );    
}

export default Button;

Button.propTypes = {
  content:PropTypes.element,
  onClick:PropTypes.func,
  className:PropTypes.string, 
  type:PropTypes.string
};

Button.defaultProps={
  type:"button",
}