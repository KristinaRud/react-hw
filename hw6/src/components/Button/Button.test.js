import Button from './Button'
import {fireEvent, render, screen} from "@testing-library/react";
import '@testing-library/jest-dom';


const handleClick = jest.fn();

describe('Button test work', () => {
	test('should Button snapshot', () => {
		const {asFragment} = render(<Button type={'submit'} className='btn-test' content="SUBMIT"/>)
		expect(asFragment()).toMatchSnapshot();
	});

	test('should handle click works', () => {
		render(<Button type={'submit'} className='btn-test' onClick={handleClick} content="SUBMIT"/>)
		const btn = screen.getByText("SUBMIT");
		fireEvent.click(btn);
		expect(handleClick).toHaveBeenCalled();
	});

	test('should have the className', () => {
		render(<Button className={'test'} content="TEST BUTTON"/>);
		const btn = screen.getByText("TEST BUTTON");

		expect(btn).toHaveClass('test');
	});

	test('should have children', () => {
		render(<Button content="TEST BUTTON"/>);
		const btn = screen.getByText("TEST BUTTON");
		expect(btn).not.toBeEmptyDOMElement();
	});
})
