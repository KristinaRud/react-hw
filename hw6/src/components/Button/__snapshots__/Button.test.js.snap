// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test work should Button snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn-test"
    type="submit"
  >
    SUBMIT
  </button>
</DocumentFragment>
`;
